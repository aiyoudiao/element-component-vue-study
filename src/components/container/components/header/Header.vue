<script setup lang="ts">
import { withDefaults } from 'vue'
interface HeaderType {
  height?: string
}

withDefaults(defineProps<HeaderType>(), {
  height: '60px',
})
</script>

<script lang="ts">
export default {
  name: 'ElHeader',
}
</script>

<template>
  <header class="el-header" :style="{ height }">
    <slot />
  </header>
</template>

<style lang="scss">
@import '../../../../theme/src/header.scss';
</style>
