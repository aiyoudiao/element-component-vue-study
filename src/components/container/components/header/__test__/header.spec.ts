import { mount } from '@vue/test-utils'
import Header from '../Header.vue'
describe('Header.vue', () => {
  it('Header snapshot', () => {
    const wrapper = mount(Header)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Header render', () => {
    const wrapper = mount(Header)
    expect(wrapper.classes()).toContain('el-header')
  })
  it('Header slot', () => {
    const wrapper = mount(Header, {
      slots: {
        default: () => 'hello world',
      },
    })
    expect(wrapper.text()).toBe('hello world')
  })
  it('Header style', () => {
    const wrapper = mount(Header, {
      props: {
        height: '90px',
      },
    })
    expect(wrapper.attributes('style')).toBe('height: 90px;')
  })
})

