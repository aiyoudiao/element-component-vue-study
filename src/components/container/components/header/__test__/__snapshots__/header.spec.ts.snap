// Vitest Snapshot v1

exports[`Header.vue > Header snapshot 1`] = `
<header
  class="el-header"
  style="height: 60px;"
>
  
  
</header>
`;
